src/compiler/emitter.ts

```diff
@@ -3269,7 +3288,12 @@ module ts {
                     startIndex = emitDirectivePrologues((<Block>node.body).statements, /*startWithNewLine*/ true);
                 }
                 var outPos = writer.getTextPos();
-                emitCaptureThisForNodeIfNecessary(node);
+
+                // In ES6, fat arrow function lexically binds this value. Therefore, when targeting es6, we can omit capturing of "this" in the fat arrow function
+                if (!isES6ArrowFunction(node)) {
+                    emitCaptureThisForNodeIfNecessary(node);

```

Comment:
why not have emitCaptureForThisNodeIfNecessary check if it's an ES6 arrow function or not?  I would think that would fall under hte bucket of 'if necessary'.